### ISSUE TITLE
[feat] Expose methods to get cursor_position

### ISSUE THREAD

**Issue by Hacksore** [tauri-apps/tauri#9250]
### Describe the problem

I have a use case where I'd like to be able to detect if the mouse intersects with a tauri window.

The reason is that I use `set_ignore_cursor_events(true)` to create an overlay displaying something useful to a user.

However, I'd like to be able to allow the user to move their mouse over the window and control the opacity so they can see what behind the window.

Here is an MVP repo that shows you can use javascript to do this but only if the window is in focus.

https://github.com/Hacksore/mouse-intersection-when-ignore-mouse-events

### Describe the solution you'd like

A way in rust to listen globally for the mouse intersecting events of windows.

- intersecting (called once)
- no longer intersecting (called once)

These events would let you run some "callback" rust code (unsure if that's rusty) but for my use case I'd want to toggle some body class on the app.

tl;dr `set_ignore_cursor_events(true)`, you mouse over a tauri window you can run rust code on enter/exit.

### Additional context

As mentioned initially you can do this in JS but it requires the window to be focused.

Demo:

https://github.com/tauri-apps/tauri/assets/996134/cae75ec0-682d-4234-86f7-bac88352547c

**Comment by amrbashir** [comment]
I think this could be solved by polling the current cursor position and do the intersection calculation on your end.
